### PR TITLE
Pass correct api.Module to host functions

### LINKED ISSUE #1211: Executing a Function from another Module
Executing a host function from a function called from another module returns a pointer to original module. Memory address referenced from the other module is not accessible, as the pointer is from the original module.

```go
//go:wasm-module host
//export printModuleName()
func printModuleName()


//export function1
func function1() {
	printModuleName()
}
```

```go
//go:wasm-module module1
//export function1
func function1()

func function2() {
	function1()
}
```

```go
func printModuleName(ctx, module) error {
	fmt.Println(module.Name())
}

````

calling function2 from module1
```go
module2.ExportedFunction("function2").Call()
```

prints

```
module2
```

Thus memory addressed in module1 are not accessible from function1 being called by function2. 

Not sure if this is intended or not.

### PR DESCRIPTION
**Pull request by mathetake**
Fixes #1211 

Previously, host functions are getting `api.Module` for the "originating" module, which is the module for api.Function currently invoked, except that the api.Module is modified withMemory for the caller's memory instance, therefore there haven't been no problem. The only issues were the methods except `Memory()` of api.Module, and this commit fixes them.

### CODE REVIEW

**Comment by mathetake** [comment]
will run 2hr fuzzing. I will make it ready after it finishes.

**Comment by mathetake** [comment]
both on amr64 and amd64 of course!

**Comment by mathetake** [comment]
arm64

File: internal/engine/compiler/compiler_controlflow_test.go [hunk 1]
```diff
@@ -17,13 +17,21 @@ func TestCompiler_compileHostFunction(t *testing.T) {
 	err := compiler.compileGoDefinedHostFunction()
 	require.NoError(t, err)
 
-	// Generate and run the code under test.
+	_, _, callerFuncLoc := compiler.runtimeValueLocationStack().getCallFrameLocations(&wasm.FunctionType{})
```
Comment (codefromthecrypt) [inline internal/engine/compiler/compiler_controlflow_test.go:20]: maybe comment on what this line is doing or if the empty ft is relevant

File: internal/engine/compiler/engine.go [hunk 2]
```diff
@@ -925,7 +929,7 @@ entry:
 			fn := calleeHostFunction.parent.goFunc
 			switch fn := fn.(type) {
 			case api.GoModuleFunction:
-				fn.Call(ce.ctx, callCtx.WithMemory(ce.memoryInstance), stack)
+				fn.Call(ce.ctx, ce.callerFunctionInstance.Module.CallCtx.WithMemory(ce.memoryInstance), stack)
```
Comment (codefromthecrypt) [inline internal/engine/compiler/engine.go:932]: due to the chain here, feels like a comment is needed, maybe extracting as a variable.

**Review by codefromthecrypt — APPROVED** [review]
Looks good and fast turnaround. maybe a bit of a summary in the PR desc

**Pull request merged**